reframe/frontend/filters.py

```diff
@@ -49,7 +49,12 @@ def have_any_name(names):
     regex_matches = []
     for n in names:
         if has_compact_names and '@' in n:
-            exact_matches.append(n.replace('@', '_'))
+            test_name, sep, id = n.rpartition('@')
+            if id.isdigit():
+                exact_matches.append((test_name, int(id)))
+            else:
+                exact_matches.append((test_name, id))
```

Comment:
Why not in the pythonic way?

```suggestion
            try:
                exact_matches.append((test, int(id)))
             except ValueError:
                exact_matches.append((test, id))
```

Reply:
The only reason to keep non-digit ids was for the fixtures so I removed the else